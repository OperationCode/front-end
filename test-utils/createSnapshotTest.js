/* eslint-env jest */
import renderer from 'react-test-renderer';

/**
 * Used to create fully-rendered snapshot test.
 * Useful for reusable component smoke tests.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const tree = renderer.create(Component).toJSON();
  expect(tree).toMatchSnapshot();
};
