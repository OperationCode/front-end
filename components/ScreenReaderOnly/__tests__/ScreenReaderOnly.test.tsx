import createSnapshotTest from 'test-utils/createSnapshotTest';
import { composeStory } from '@storybook/react';
import meta, { Default } from '../__stories__/ScreenReaderOnly.stories';

const ScreenReaderOnlyStory = composeStory(Default, meta);

describe('ScreenReaderOnly', () => {
  it('should render with required props', () => {
    createSnapshotTest(<ScreenReaderOnlyStory {...Default.args} />);
  });
});
