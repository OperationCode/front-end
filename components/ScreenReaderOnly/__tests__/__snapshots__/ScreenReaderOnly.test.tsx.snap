// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ScreenReaderOnly > should render with required props 1`] = `
[
  <span>
    Content never displayed, but it is rendered
  </span>,
  <span
    className="sr-only"
    data-testid="SCREEN_READER_ONLY"
  >
    ScreenReader content
  </span>,
]
`;
